### ISSUE TITLE
[X86] error in backend: Cannot emit physreg copy instruction

### ISSUE THREAD

**Issue by ramosian-glider** [llvm/llvm-project#62625]
(also reported at https://github.com/ClangBuiltLinux/linux/issues/1845)

When compiling the reproducer extracted from the Linux kernel v6.4-rc1, trunk Clang crashes in `tryOptimizeLEAtoMOV()`:

```console
$ clang -v
clang version 17.0.0 (git@github.com:llvm/llvm-project.git feb69dad8a131d718d7c44e983e1cf6ef202127c)
Target: x86_64-unknown-linux-gnu
Thread model: posix
```

```console
$ cat vmwgfx.i
static void vmw_close_channel(short *channel_id)
{
	unsigned long eax, ebx, ecx, edx, si, di;
	({
		asm volatile(""
			     : "=a"(eax), "=b"(ebx), "=c"(ecx), "=d"(edx),
			       "=S"(si), "=D"(di)
			     : "a"(0), "b"(0), "c"(0), "d"(*channel_id),
			       "S"(si), "D"(di)
			     : "memory");
	});
}
void vmw_host_printf(const char *fmt, ...)
{
	__builtin_va_list ap;
	short id;
	__builtin_va_start(ap, fmt);
	vmw_close_channel(&id);
}
```

```console
$ clang -mstack-alignment=8 -O2 -fsanitize=kernel-memory -c vmwgfx.i
fatal error: error in backend: Cannot emit physreg copy instruction
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.Th
Stack dump:
0.	Program arguments: /usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang -mstack-alignment=8 -O2 -fsanitize=kernel-memory -c vmwgfx.i
1.	<eof> parser at end of file
2.	Code generation
3.	Running pass 'Function Pass Manager' on module 'vmwgfx.i'.
4.	Running pass 'Argument Stack Rebase' on function '@vmw_host_printf'
 #0 0x000055ff98150efb llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x39f2efb)
 #1 0x000055ff9814e8ab llvm::sys::RunSignalHandlers() (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x39f08ab)
 #2 0x000055ff980a5c76 llvm::CrashRecoveryContext::HandleExit(int) (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x3947c76)
 #3 0x000055ff98148e9e llvm::sys::Process::Exit(int, bool) (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x39eae9e)
 #4 0x000055ff956b4214 LLVMErrorHandler(void*, char const*, bool) cc1_main.cpp:0:0
 #5 0x000055ff980b1489 llvm::report_fatal_error(llvm::Twine const&, bool) (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x3953489)
 #6 0x000055ff980b15fa (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x39535fa)
 #7 0x000055ff96eefe0f llvm::X86InstrInfo::copyPhysReg(llvm::MachineBasicBlock&, llvm::MachineInstrBundleIterator<llvm::MachineInstr, false>, llvm::DebugLoc const&, llvm::MCRegister, llvm::MCRegister, bool) const (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x2791e0f)
 #8 0x000055ff96c42b57 tryOptimizeLEAtoMOV(llvm::MachineInstrBundleIterator<llvm::MachineInstr, false>) X86RegisterInfo.cpp:0:0
 #9 0x000055ff96c42dd8 llvm::X86RegisterInfo::eliminateFrameIndex(llvm::MachineInstrBundleIterator<llvm::MachineInstr, false>, unsigned int, llvm::Register, int) const (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x24e4dd8)
#10 0x000055ff96c7acc6 (anonymous namespace)::X86ArgumentStackSlotPass::runOnMachineFunction(llvm::MachineFunction&) X86ArgumentStackSlotRebase.cpp:0:0
#11 0x000055ff9752e41b llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (.part.0) MachineFunctionPass.cpp:0:0
#12 0x000055ff97ad9639 llvm::FPPassManager::runOnFunction(llvm::Function&) (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x337b639)
#13 0x000055ff97ad9771 llvm::FPPassManager::runOnModule(llvm::Module&) (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x337b771)
#14 0x000055ff97ada077 llvm::legacy::PassManagerImpl::run(llvm::Module&) (/usr/local/google/src/llvm-git-monorepo/llvm_cmake_build/bin//clang+0x337c077)
```

**Comment by phoebewang** [comment]
cc @LuoYuanke

**Comment by LuoYuanke** [comment]
Thanks phoebe for notification. I'll take a look at this issue.

**Comment by LuoYuanke** [comment]
Candidate patch https://reviews.llvm.org/D150521.

**Comment by RKSimon** [comment]
This landed at cf59d67c6407bc27693a08e18ad9b8254a38ec2d - @ramosian-glider please can you confirm the fix?

**Comment by phoebewang** [comment]
Do we need to backport it?
